apps/dokploy/components/dashboard/docker/logs/docker-logs-id.tsx

```diff
@@ -1,110 +1,158 @@
+		const wsUrl = `${protocol}//${window.location.host}/docker-container-logs?containerId=${containerId}&tail=${lines}${serverId ? `&serverId=${serverId}` : ""}&timeFilter=${timeFilter}`;
 
-		const wsUrl = `${protocol}//${window.location.host}/docker-container-logs?containerId=${containerId}&tail=${lines}${serverId ? `&serverId=${serverId}` : ""}`;
 		const ws = new WebSocket(wsUrl);
 		wsRef.current = ws;
-		const fitAddon = new FitAddon();
-		termi.loadAddon(fitAddon);
-		// @ts-ignore
-		termi.open(container);
-		fitAddon.fit();
-		termi.focus();
-		setTerm(termi);
-
-		ws.onerror = (error) => {
-			console.error("WebSocket error: ", error);
-		};
 
-		ws.onmessage = (e) => {
-			termi.write(e.data);
+		ws.onmessage = ({ data }) => {
+			const formattedLog = formatLogLine(data);
+			if (searchTerm.length <= 1 || data.includes(searchTerm)) {
+				terminal.write(`${formattedLog}\r\n`);
+			}
 		};
 
 		ws.onclose = (e) => {
-			console.log(e.reason);
-
-			termi.write(`Connection closed!\nReason: ${e.reason}\n`);
+			terminal.write(`Connection closed!\nReason: ${e.reason}\n`);
 			wsRef.current = null;
 		};
-		return () => {
-			if (wsRef.current?.readyState === WebSocket.OPEN) {
-				ws.close();
-				wsRef.current = null;
-			}
-		};
-	}, [lines, containerId]);
+
+		return () => ws.close();
+	}, [lines, containerId, timeFilter, searchTerm, id, serverId]);
 
 	useEffect(() => {
 		term?.clear();
 	}, [lines, term]);
 
 	return (
 		<div className="flex flex-col gap-4">
-			<div className="flex flex-col gap-2">
-				<Label>
-					<span>Number of lines to show</span>
-				</Label>
-				<Input
-					type="text"
-					placeholder="Number of lines to show (Defaults to 35)"
-					value={lines}
-					onChange={(e) => {
-						setLines(Number(e.target.value) || 1);
-					}}
-				/>
+			<div className="flex items-center gap-2 bg-background/95 rounded-md">
+				<div className="flex flex-col gap-1">
+					<Label htmlFor="time-filter">Lines & Time Range</Label>
+					<Select
+						value={timeFilter}
+						onValueChange={(value: TimeFilter) => {
+							setTimeFilter(value);
+							if (value.endsWith("lines")) {
+								setLines(Number.parseInt(value));
+							}
+						}}
+					>
+						<SelectTrigger id="time-filter" className="w-[160px]">
+							<SelectValue placeholder="Select filter" />
+						</SelectTrigger>
+						<SelectContent>
+							<SelectItem value="50lines">50 lines</SelectItem>
+							<SelectItem value="150lines">150 lines</SelectItem>
+							<SelectItem value="500lines">500 lines</SelectItem>
+							<SelectItem value="1000lines">1000 lines</SelectItem>
+							<SelectSeparator className="my-2" />
+							<SelectItem value="last10min">Last 10 min</SelectItem>
+							<SelectItem value="last30min">Last 30 min</SelectItem>
+							<SelectItem value="lasthour">Last hour</SelectItem>
+							<SelectItem value="last24h">Last 24h</SelectItem>
+						</SelectContent>
```

Comment:
How did you deteremine the time of the logs? I don't see any logic applied to the logs itself in the backend? can you explain how the search and filters works?

Reply:
@Siumauricio I'm not sure how they're implementing it, but this can be set with the `--since` switch on `docker logs`. [Documentation](https://docs.docker.com/reference/cli/docker/container/logs/#options)

example: `docker logs --follow --since 30m wattpadarchive-workers`

Reply:
> I don't see any logic applied to the logs itself in the backend?

Me neither, if the websocket doesn't support `&timeFilter`, that logic could be added to
```js
ws.onmessage = ({ data }) => {
			const formattedLog = formatLogLine(data);
			if (searchTerm.length <= 1 || data.includes(searchTerm)) {
				terminal.write(`${formattedLog}\r\n`);
			}
		};
```
as well. Kinda janky though, when `docker logs --since` exists already.